catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/sharing/ShareableMetacardImpl.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.sharing;
+
+import com.google.common.collect.Sets;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.types.SecurityAttributes;
+import ddf.catalog.data.types.Core;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** A metacard that supports sharing. */
+public class ShareableMetacardImpl extends MetacardImpl {
+  private static MetacardType metacardType = null;
+
+  public ShareableMetacardImpl(MetacardType type) {
+    super(type);
+  }
+
+  public ShareableMetacardImpl(Metacard metacard) {
+    super(metacard);
+  }
+
+  public ShareableMetacardImpl(Metacard metacard, MetacardType type) {
+    super(metacard, type);
+  }
+
+  public ShareableMetacardImpl() {
+    super(metacardType);
+  }
+
+  public ShareableMetacardImpl(String id) {
+    this();
+    setId(id);
+  }
+
+  public static void setShareableMetacardType(MetacardType metacardType) {
+    ShareableMetacardImpl.metacardType = metacardType;
+  }
+
+  /**
+   * Check if a given metacard is a shareable metacard by checking the tags metacard attribute.
+   *
+   * @param metacard the metacard to check.
+   * @return true if the provided metacard is a shareable metacard, false otherwise.
+   */
+  public static boolean isShareableMetacard(Metacard metacard) {
```

Comment:
❓ Should this be renamed to `canShare` since we're moving to a model where most (if not all) metacards support sharing as long as they have security attributes? Also, for the time being, `ShareableMetacard` might be more like a "decorator" which provides additional operations upon an injected metacard.